packages/integration-tests/src/setup_package_manager.ts

```diff
@@ -0,0 +1,115 @@
+import { execa } from 'execa';
+import * as fsp from 'fs/promises';
+import { existsSync } from 'fs';
+import * as path from 'path';
+import * as os from 'os';
+
+const customRegistry = 'http://localhost:4873';
+
+// TODO: refactor into `type PackageManagerInitializer` and have sub-types with a factory.
+export type PackageManager = 'npm' | 'yarn-classic' | 'yarn-modern' | 'pnpm';
+export type PackageManagerExecutable = 'npx' | 'yarn' | 'pnpm';
+
+const initializeNpm = async () => {
+  const { stdout } = await execa('npm', ['config', 'get', 'cache']);
+  const npxCacheLocation = path.join(stdout.toString().trim(), '_npx');
+  if (existsSync(npxCacheLocation)) {
+    await fsp.rm(npxCacheLocation, { recursive: true });
+  }
+};
+
+const initializePnpm = async () => {
+  const packageManager = 'pnpm';
+  await execa('npm', ['install', '-g', packageManager], {
+    stdio: 'inherit',
+  });
+  await execa(packageManager, ['--version']);
```

Comment:
shouldn't we be using `runPackageManager` for these commands?

Reply:
Not needed since `packageManager = 'pnpm'` in this case